provision/core/vvv/provision.sh

```diff
@@ -96,8 +96,21 @@ function shyaml_setup() {
   # Used for passing custom parameters to the bash provisioning scripts
   if [ ! -f /usr/local/bin/shyaml ]; then
     vvv_info " * Installing Shyaml for bash provisioning.."
-    sudo pip3 install wheel
-    sudo pip3 install shyaml
+
+    local OSVERSION_NUMBER=$(lsb_release lsb_release -sr)
+
+    # Ubuntu 24 making it hard to install pip packages, throwing externally-managed-environment error
+    # https://stackoverflow.com/a/75722775
+    if dpkg --compare-versions "${OSVERSION_NUMBER[@]}" ge "24.04"
+    then
+      # to make it available globally this is the last workaround, hopefully it doesn't break the system
+      # TODO: try to find a better alternative way to install
```

Comment:
is there somewhere I can read up on why this happens or where the fix came from? Would be good to have it in the comments, I've created a separate PR for shyaml fixes

Reply:
- https://peps.python.org/pep-0668/
- https://pimylifeup.com/python-externally-managed-environment/
- https://stackoverflow.com/questions/75608323/how-do-i-solve-error-externally-managed-environment-every-time-i-use-pip-3/75722775#75722775